third_party/terraform/utils/transport.go

```diff
@@ -53,8 +60,21 @@ func sendRequest(config *Config, method, rawurl string, body map[string]interfac
 		return nil, err
 	}
 	req.Header = reqHeaders
+
+	if timeout != 0 {
+		ctx, cancel := context.WithTimeout(context.TODO(), timeout)
```

Comment:
From my understanding of the docs, context.Background is meant for background tasks that are never cancelled. So in this case since we're intentionally going to be calling .cancel to terminate the request I don't think we can use it. Apparently context.TODO is actually encouraged:

> Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).

Reply:
awesome, thanks for the explanation! can that explanation become a comment?